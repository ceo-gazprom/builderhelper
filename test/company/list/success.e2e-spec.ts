import { Test, TestingModule } from '@nestjs/testing';
import type { INestApplication } from '@nestjs/common';
import * as request from 'supertest';
import { DataSource } from 'typeorm';
import { clearDatabase } from '../../test-utils';
import { AppModule } from '../../../src/app.module';

describe('/companies/list (GET) - список компаний', () => {
  let app: INestApplication;

  beforeAll(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
    await clearDatabase(app.get(DataSource));
  });

  afterAll(async () => {
    await app.close();
  });

  it('Должен вернуть список компаний с пагинацией', async () => {
    await request(app.getHttpServer())
      .post('/companies/register')
      .send({
        name: 'Test Company',
        email: 'testcompany@example.com',
        password: 'password123',
      })
      await request(app.getHttpServer())
      .post('/companies/register')
      .send({
        name: 'Test Company 1',
        email: 'testcompany1@example.com',
        password: 'password123',
      })
      await request(app.getHttpServer())
      .post('/companies/register')
      .send({
        name: 'Test Company 2',
        email: 'testcompany2@example.com',
        password: 'password123',
      })
      .expect(201);
  
    const response = await request(app.getHttpServer())
      .get('/companies/list')
      .expect(200);

    expect(response.body).toMatchSnapshot();
  });
});