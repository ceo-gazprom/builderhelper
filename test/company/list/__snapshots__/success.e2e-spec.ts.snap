// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/company/list (GET) - список компаний Должен вернуть список компаний с пагинацией 1`] = `
[
  {
    "email": "testcompany@example.com",
    "id": 1,
    "name": "Test Company",
  },
  {
    "email": "testcompany1@example.com",
    "id": 2,
    "name": "Test Company 1",
  },
  {
    "email": "testcompany2@example.com",
    "id": 3,
    "name": "Test Company 2",
  },
]
`;
